### ISSUE TITLE
Webviews not rendering on OSX

### ISSUE THREAD

**Issue by kontrollanten** [brrd/electron-tabs#22]
I have the following code:
```
<div class="flex-container">
	<div class="etabs-tabgroup
		<div class="etabs-tabs"></div>
		<div class="etabs-buttons"></div>
	</div>
	<div class="etabs-views"></div>
</div>
```
```
.flex-container {
    display: flex;
}
```
```
    const TabGroup = require('electron-tabs');
    
    let tabGroup = new TabGroup();
    let tab = tabGroup.addTab({
      title: "my@email.com",
      src: "https://mail.protonmail.com/login",
      visible: true
    });
    tab.activate();
    let tab2 = tabGroup.addTab({
      title: "my.second@email.com",
      src: "https://mail.protonmail.com/login",
      visible: true
    });
```

When does it *not* render:
* With the code above.
* If I create a custom webview inside the flex-container, none of the webviews will ender.
* With both of the following but also remove all electron-tabs appended stylesheets.

When does it render:
* If I, in the developer tools, move the webview outside the etabs-views div it'll start loading and when I move it back it displays correctly.
* If I create a custom webview inside the flex-container and remove the electron-tabs module.
* If I just create a webview *outside* the flex-container, then all webviews (and all the tabs) will load correctly.

electron 1.6.6
electron-tabs 0.6.0
OS: Mac OS 10.12.4

I found the following https://github.com/electron/electron/issues/8505 and thought it may be correlated?

**Comment by brrd** [comment]
Would you mind writing a minimal example of this issue and send it here as a .zip archive? This would be really helpful.

**Comment by kontrollanten** [comment]
You can check this repo https://github.com/kontrollanten/Protonmail/tree/enable-multiple-accounts 

The HTML is in app/index.html and tabs initiation is in src/browser.js. Now that I'm writing it just hit me that maybe the error is because I'm initiating the tabs after DOMContentReady.

**Comment by brrd** [comment]
Could you please provide a minimal app that reproduces this issue and does nothing else? It's the best way to be sure we avoid side effects while reproducing the issue.

**Comment by kontrollanten** [comment]
Yes, I'll get back to you with that

**Comment by kontrollanten** [comment]
https://github.com/kontrollanten/electron-tabs-not-rendering When I remove ``tab.activate();`` the second tab get rendered on click, but not the first one. I'm sorry but I don't have time investigate it further right now But maybe you can see what's going wrong?

**Comment by brrd** [comment]
Thanks for this sample app. 

Everything works well on my Linux (Ubuntu 14.04) so I guess it is platform specific and probably related to electron/electron#8505 as you suggested. So we should probably try the workarounds which are mentioned in this thread (using `visibility: hidden` sounds like a good option).

The computer I have at the moment is not powerful enough to run OSX in a VM so I currently have no way to reproduce and fix this issue. But I will probably be able to do it in some weeks (if no one fixes it before).

**Comment by kontrollanten** [comment]
Thanks your quick replies! It seems that if I move the styles to inline everything works as excepted. So I guess this is an electron issue rather than electron tabs issue.

**Comment by brrd** [comment]
Well, I should have RTFM again: https://electron.atom.io/docs/api/webview-tag/#css-styling-notes :confused:

**Comment by avatar21** [comment]
I got it working, well kinda, by using following code ... but still i have to wait ...

```json
let tabGroup = new TabGroup()
let tabHome = tabGroup.addTab({
        title: "home",
        visible: true,
        src: "http://electron.atom.io",
        closable: false,
        active: true
    })
let tabSettings = tabGroup.addTab({
        title: "settings",
        visible: true,
        src: "http://www.yahoo.com",
        closable: false,
        active: true
    })
tabGroup.on("tab-added", (tab, tabGroup) => {
        if (tab === tabHome) {
            tabHome.active()
            let webview = tabHome.webview
            webview.loadURL("http://electron.atom.io")
        }
    });
```

Meanwhile, I still found that it could be linked to this issue:
https://www.chromestatus.com/feature/5718547946799104

**Comment by brrd** [comment]
I tried to fix this in commit 0b078d235df70601ab5d4561b8bbad0890e706c5 but since @kontrollanten deleted his sample (https://github.com/kontrollanten/electron-tabs-not-rendering) I can not reproduce this issue anymore. 

Can someone tell me if this issue is fixed in version 0.7.2 on OSX?

**Comment by kontrollanten** [comment]
I'm sorry @brrd The sample was deleted upon my yearly cleanup. I'll see if I can get another sample back up again. @avatar21 Can you publish a sample where it's not working? And then try with the latest update from @brrd ?

**Comment by mike-mccormick** [comment]
Hey,

I'm seeing this issue too.

The build I'm running has @brrd's fixes, but this still doesn't work sadly.

I'm currently using this within my renderrer.js:

```const TabGroup = require("electron-tabs");

let tabGroup = new TabGroup();
let tabHome = tabGroup.addTab({
    title: "Home",
    src: "http://electron.atom.io",
    visible: true,
    closable: false,
    active: true
});

let tabSupport = tabGroup.addTab({
    title: 'Google',
    src: 'http://www.google.com',
    visible: true,
    closable: false,
    active: true
});
```

**Comment by mike-mccormick** [comment]
Toying with this, 

If I drop line 11 from index.js entirely ( `visibility: hidden;`) it seems to resolve the issue for me.

@avatar21 or @kontrollanten ontrollanten - what do you guys get trying this?

Just so we're on the same page, line 7-23 for me are currently:

```// Inject styles
(function () {
    const styles = `
        webview {
            width: 0px;
            height: 0px;
        }
        webview.visible {
            visibility: visible;
            width: 100%;
            height: 100%;
            top: 0;
            right: 0;
            bottom: 0;
            left: 0;
        }
    `;
```

**Comment by regular** [comment]
@mike-mccormick yes, deleting that line fixed it for us, too.

**Comment by brrd** [comment]
@mike-mccormick @regular Fixed in version 0.9.1
